### PR TITLE
[PDI-18031] - ODBC access type does not work and should be removed from connection dialogs

### PR DESCRIPTION
**Pull request by joana-fb**

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 26m 47s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
core,dbdialog,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 2 test(s) failed:

<details><summary><b>org.pentaho.di.core.database.MSAccessDatabaseMetaTest.testSettings</b> (click to expand)</summary><p>

```
array lengths differed, expected.length=1 actual.length=0
```

</p></details>

<details><summary><b>org.pentaho.ui.database.event.FragmentHandlerTest.org.pentaho.ui.database.event.FragmentHandlerTest</b> (click to expand)</summary><p>

```
No more handles [gtk_init_check() failed]
```

</p></details>

  
Tests run: **4119**, Failures: **2**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9484/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 38m 9s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
core,dbdialog,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.ui.database.event.FragmentHandlerTest.org.pentaho.ui.database.event.FragmentHandlerTest</b> (click to expand)</summary><p>

```
No more handles [gtk_init_check() failed]
```

</p></details>

  
Tests run: **4119**, Failures: **1**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9484/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 45m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.ui.database.event.FragmentHandlerTest.org.pentaho.ui.database.event.FragmentHandlerTest</b> (click to expand)</summary><p>

```
No more handles [gtk_init_check() failed]
```

</p></details>

  
Tests run: **5510**, Failures: **1**, Skipped: **23** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9484/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 30m 51s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
core,dbdialog,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.ui.database.event.FragmentHandlerTest.org.pentaho.ui.database.event.FragmentHandlerTest</b> (click to expand)</summary><p>

```
No more handles [gtk_init_check() failed]
```

</p></details>

  
Tests run: **4112**, Failures: **1**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9484/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 29m 36s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
core,dbdialog,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.ui.database.event.FragmentHandlerTest.org.pentaho.ui.database.event.FragmentHandlerTest</b> (click to expand)</summary><p>

```
No more handles [gtk_init_check() failed]
```

</p></details>

  
Tests run: **4119**, Failures: **1**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9484/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 23m 57s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
core,dbdialog,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.ui.database.event.FragmentHandlerTest.org.pentaho.ui.database.event.FragmentHandlerTest</b> (click to expand)</summary><p>

```
No more handles [gtk_init_check() failed]
```

</p></details>

  
Tests run: **4119**, Failures: **1**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9484/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/di/core/database/RemedyActionRequestSystemDatabaseMeta.java [hunk 1]
```diff
@@ -33,7 +33,7 @@
 public class RemedyActionRequestSystemDatabaseMeta extends GenericDatabaseMeta implements DatabaseInterface {
   @Override
   public int[] getAccessTypeList() {
-    return new int[] { DatabaseMeta.TYPE_ACCESS_ODBC, DatabaseMeta.TYPE_ACCESS_JNDI };
+    return new int[] {DatabaseMeta.TYPE_ACCESS_JNDI };
```
Comment (smmribeiro) [inline core/src/main/java/org/pentaho/di/core/database/RemedyActionRequestSystemDatabaseMeta.java:36]: Considering the fact that "getDriverClass" and "getURL" methods return ODBC-only results; seems that Remedy Action Request System Database has to also be deprecated (and later dropped) similar to what is being done to "MS Access" and "dBase *".

File: core/src/test/java/org/pentaho/di/core/database/RemedyActionRequestSystemDatabaseMetaTest.java [hunk 2]
```diff
@@ -34,12 +34,11 @@ public class RemedyActionRequestSystemDatabaseMetaTest {
   @Before
   public void setupBefore() {
     odbcMeta = new RemedyActionRequestSystemDatabaseMeta();
-    odbcMeta.setAccessType( DatabaseMeta.TYPE_ACCESS_ODBC );
   }
 
   @Test
   public void testSettings() throws Exception {
-    assertArrayEquals( new int[] { DatabaseMeta.TYPE_ACCESS_ODBC, DatabaseMeta.TYPE_ACCESS_JNDI },
+    assertArrayEquals( new int[] { DatabaseMeta.TYPE_ACCESS_JNDI },
```
Comment (smmribeiro) [inline core/src/test/java/org/pentaho/di/core/database/RemedyActionRequestSystemDatabaseMetaTest.java:41]: Check review for "RemedyActionRequestSystemDatabaseMeta.java" class: this test class should be marked as deprecated and later removed.

File: ui/src/main/java/org/pentaho/di/ui/core/database/wizard/CreateDatabaseWizard.java [hunk 3]
```diff
@@ -71,8 +71,6 @@ public DatabaseMeta createAndRunDatabaseWizard( Shell shell, PropsUI props, List
 
     final CreateDatabaseWizardPageOCI pageoci = new CreateDatabaseWizardPageOCI( "oci", props, newDBInfo );
 
-    final CreateDatabaseWizardPageODBC pageodbc = new CreateDatabaseWizardPageODBC( "odbc", props, newDBInfo );
```
Comment (smmribeiro) [inline ui/src/main/java/org/pentaho/di/ui/core/database/wizard/CreateDatabaseWizard.java:74]: This 'CreateDatabaseWizardPageODBC' class should also be removed from code!
It exists here: [ui/src/main/java/org/pentaho/di/ui/core/database/wizard](https://github.com/pentaho/pentaho-kettle/tree/master/ui/src/main/java/org/pentaho/di/ui/core/database/wizard)

Don't forget to also remove the messages referring to this same class - check message files here: [ui/src/main/resources/org/pentaho/di/ui/core/database/wizard/messages](https://github.com/pentaho/pentaho-kettle/tree/master/ui/src/main/resources/org/pentaho/di/ui/core/database/wizard/messages)

File: ui/src/main/java/org/pentaho/di/ui/repository/kdr/KettleDatabaseRepositoryDialog.java [hunk 4]
```diff
@@ -435,20 +435,6 @@ private void create() {
     getInfo( repositoryMeta );
 
     if ( repositoryMeta.getConnection() != null ) {
-      if ( repositoryMeta.getConnection().getAccessType() == DatabaseMeta.TYPE_ACCESS_ODBC ) {
-        // Show a warning: using ODBC is not always the best choice ;-)
-        System.out.println( "Show ODBC warning..." );
-
-        MessageBox qmb = new MessageBox( shell, SWT.ICON_WARNING | SWT.YES | SWT.NO );
-        qmb.setMessage( BaseMessages.getString( PKG, "RepositoryDialog.Dialog.ODBCIsNotSafe.Message", Const.CR,
```
Comment (smmribeiro) [inline ui/src/main/java/org/pentaho/di/ui/repository/kdr/KettleDatabaseRepositoryDialog.java:443]: Please remove the "RepositoryDialog.Dialog.ODBCIsNotSafe.*" entries from the message files here: [ui/src/main/resources/org/pentaho/di/ui/repository/dialog/messages](https://github.com/pentaho/pentaho-kettle/tree/master/ui/src/main/resources/org/pentaho/di/ui/repository/dialog/messages)

**Review by smmribeiro — CHANGES_REQUESTED** [review]
Please check my comments.

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>    High | org.jdom:jdom:1.1.3 | org.jdom:jdom 1.1.3 | - | CVE-2021-33813 |

</div>

### 🔬 Research Details
**Description:**
An XXE issue in SAXBuilder in JDOM through 2.0.6 allows attackers to cause a denial of service via a crafted HTTP request.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
<div align='center'>

[🚨 Frogbot scanned this pull request and found the below:](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>


## 📦 Vulnerable Dependencies
### ✍️ Summary
<div align='center'>

| SEVERITY                | DIRECT DEPENDENCIES                  | IMPACTED DEPENDENCY                  | FIXED VERSIONS                  | CVES                  |
| :---------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: | :-----------------------------------: |
| <br>    High | org.jdom:jdom:1.1.3 | org.jdom:jdom 1.1.3 | - | CVE-2021-33813 |

</div>

### 🔬 Research Details
**Description:**
An XXE issue in SAXBuilder in JDOM through 2.0.6 allows attackers to cause a denial of service via a crafted HTTP request.
<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :x: Build failed in 39m 36s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
core,dbdialog,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.ui.database.event.FragmentHandlerTest.org.pentaho.ui.database.event.FragmentHandlerTest</b> (click to expand)</summary><p>

```
No more handles [gtk_init_check() failed]
```

</p></details>

  
Tests run: **4119**, Failures: **1**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9484/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 46m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.ui.database.event.FragmentHandlerTest.org.pentaho.ui.database.event.FragmentHandlerTest</b> (click to expand)</summary><p>

```
No more handles [gtk_init_check() failed]
```

</p></details>

  
Tests run: **5510**, Failures: **1**, Skipped: **23** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9484/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 26m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
core,dbdialog,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.ui.database.event.FragmentHandlerTest.org.pentaho.ui.database.event.FragmentHandlerTest</b> (click to expand)</summary><p>

```
No more handles [gtk_init_check() failed]
```

</p></details>

  
Tests run: **4119**, Failures: **1**, Skipped: **14** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9484/9/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by befc — APPROVED** [review]

**Pull request merged**